### ISSUE TITLE
ConcurrentModificationException for display-plugin-updates on Maven 3.3.9

### ISSUE THREAD

**Issue by mattnelson** [mojohaus/versions-maven-plugin#40]
When executing a parallel build with Maven 3.3.9 and the display-plugin-updates goal a ConcurrentModificationException is thrown.

```
mvn -T 1C versions:display-plugin-updates
```

```
[ERROR] Failed to execute goal org.codehaus.mojo:versions-maven-plugin:2.2:display-plugin-updates (default-cli) on project test-project: Execution default-cli of goal org.codehaus.mojo:versions-maven-plugin:2.2:display-plugin-updates failed. ConcurrentModificationException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.codehaus.mojo:versions-maven-plugin:2.2:display-plugin-updates (default-cli) on project test-project: Execution default-cli of goal org.codehaus.mojo:versions-maven-plugin:2.2:display-plugin-updates failed.
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
    at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call(MultiThreadedBuilder.java:185)
    at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call(MultiThreadedBuilder.java:181)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-cli of goal org.codehaus.mojo:versions-maven-plugin:2.2:display-plugin-updates failed.
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:145)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
    ... 11 more
Caused by: java.util.ConcurrentModificationException
    at java.util.ArrayList$Itr.checkForComodification(ArrayList.java:901)
    at java.util.ArrayList$Itr.next(ArrayList.java:851)
    at org.apache.maven.model.io.xpp3.MavenXpp3Writer.writeBuild(MavenXpp3Writer.java:328)
    at org.apache.maven.model.io.xpp3.MavenXpp3Writer.writeModel(MavenXpp3Writer.java:1115)
    at org.apache.maven.model.io.xpp3.MavenXpp3Writer.write(MavenXpp3Writer.java:100)
    at org.codehaus.mojo.versions.DisplayPluginUpdatesMojo.execute(DisplayPluginUpdatesMojo.java:419)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
    ... 12 more
```

**Comment by diroussel** [comment]
I've upgrade to maven 3.3.9 and see another similar problem.  When I build with "mvn clean install" by project builds ok, but when I build with "mvn -T4 clean install", I get random compilation errors.  Either package does not exist, or "cannot find symbol".  The source code is valid.

Must be some threading related change in the release that are unsafe.

**Comment by khmarbaise** [comment]
Calling `mvn -T4 clean install` you are referencing the whole build and not versions-maven-plugin in particular ?

**Comment by davidkarlsen** [comment]
Also see #54 - it is possible to trigger the problem by only running the versions plugin

**Comment by khmarbaise** [comment]
See the information #54. 
Fixed in [maven-3.5.0-beta-1](http://www.mail-archive.com/users@maven.apache.org/msg139350.html) instead...which [has been](http://www.mail-archive.com/users@maven.apache.org/msg139350.html) released...
